lib/index.js

```diff
@@ -828,8 +828,13 @@ Stream.prototype.destroy = function () {
     _(this._consumers).each(function (consumer) {
         self._removeConsumer(consumer);
     });
+    _(this._observers).each(function (observer) {
+        self._removeObserver(observer)
+    });
+
     if (this.source) {
         this.source._removeConsumer(this);
+        this.source._removeObserver(this);
```

Comment:
hrrm, since `source` was removed when creating an observer, there is no longer a backreference from the observer to the stream. 

It seems we need a backreference to remove the observer being destroyed from the stream that created it; as it's still stored in `_observers`.